Migrate auth token info - from HELP to DevPortal (#501)
aiven/devportal: docs/platform/concepts/authentication-tokens.rst

        Server-->>-Client: Your token is valid. Here's a list of your resources.

How Aiven manages user sessions using authentication tokens
-----------------------------------------------------------

Each time a user signs in to Aiven either via the web console, Aiven command line client, or direct REST API call, the server creates a new authentication token associated with the user.
These tokens are set to expire after 30 days (subject to change) but the expiry date is adjusted whenever the token is used so the token may remain valid practically indefinitely if it is used frequently enough.

Token counts
------------

The system has hard limits for how many valid authentication tokens are allowed per user. This limit is different for tokens that are created as a result of sign in operation and for tokens created explicitly. The max token count is **10** for user created tokens but the system never invalidates the tokens unless they expire or they are explicitly revoked. For automatically created tokens the limit is **1000** but when this limit is reached the system automatically deletes tokens that have been used least recently to avoid going above the limit.

> Therefore, an old token can stop working even if it hasn't expired nor been explicitly revoked. To avoid running into problems with this behavior you should always make sure you sign out after sign in instead of just discarding your authentication token. This is mostly relevant for automation which automatically signs in. The Aiven web console automatically revokes current token when signing out and the Aiven command line client also provides a sign out command (avn user logout).

Reviewer: ```suggestion
Therefore, an old token can stop working even if it hasn't expired nor been explicitly revoked. To avoid running into problems with this behavior you should always make sure you sign out after sign in instead of just discarding your authentication token. This is mostly relevant for automation which automatically signs in. The Aiven web console automatically revokes the current token when signing out and the Aiven command line client also provides a sign out command (avn user logout).
```
^ add missing "the" between `revokes` and `current token`

Authentication tokens
=====================

> An authentication token allows a user to programmatically access resources instead of using their username-password. Unlike the username-password combination that has a longer shelf life (in some case, they never expire); you can limit how long an authentication token is valid for or revoke it completely.

Reviewer 2: Should we also state that several tokens can be active at the time, so you can create a token for a specific use case and revoke when the project is finished.

Author: Great point. I'll be sure to add that line @ftisiot

.. mermaid::

> sequenceDiagram

Reviewer 2: This generates an error in the preview 
![image](https://user-images.githubusercontent.com/11272743/151139569-8c31b661-5622-4b61-8124-5681a1e971f2.png)

Author: Yeah this is the main reason I had the PR in draft. Not sure on the change to render the valid diagram.

Reviewer 2: Fixed, you needed an extra <TAB>

    Client->>+Server: Here's my username/password. Give me a token.
    Server-->>-Client: Your username/password is valid. Here's a token. 
    Client->>+Server: Here's a token. Show me a list of my resources.
    Server-->>-Client: Your token is valid. Here's a list of your resources.

How Aiven manages user sessions using authentication tokens
-----------------------------------------------------------

Each time a user signs in to Aiven either via the web console, Aiven command line client, or direct REST API call, the server creates a new authentication token associated with the user.
These tokens are set to expire after 30 days (subject to change) but the expiry date is adjusted whenever the token is used so the token may remain valid practically indefinitely if it is used frequently enough.

Token counts
------------

> The system has hard limits for how many valid authentication tokens are allowed per user. This limit is different for tokens that are created as a result of sign in operation and for tokens created explicitly; the limit for explicitly created tokens is small but the system never invalidates the tokens unless they expire or they are explicitly revoked. For automatically created tokens the limit is higher but when the limit is reached the system automatically deletes tokens that have been used least recently to avoid going above the limit.

Reviewer 2: "small" and "higher" are not really quantitative information, and might generate more questions, do we have a number that we can share?

Author: Another good point. I have asked on an internal channel to get specific numbers.

> This behavior for automatically created tokens can result in a token that hasn't expired nor been explicitly revoked to stop working. To avoid running into problems with this behavior you should always make sure you sign out after sign in instead of just discarding your authentication token. This is mostly relevant for automation which automatically signs in. The Aiven web console automatically revokes current token when signing out and the Aiven command line client also provides a sign out command (avn user logout).

Reviewer 2: ```suggestion
Therefore, an old token can stop working even if it hasn't expired nor been explicitly revoked. To avoid running into problems with this behavior you should always make sure you sign out after sign in instead of just discarding your authentication token. This is mostly relevant for automation which automatically signs in. The Aiven web console automatically revokes current token when signing out and the Aiven command line client also provides a sign out command (avn user logout).
```

Also, should this be a Warning?

Note about old authentication tokens
------------------------------------

> The sign in API call used to hand out tokens that were not explicitly tracked and couldn't be individually revoked (prior to 2018-06-25). These old tokens remain valid and become automatically tracked whenever they're used so that they can be revoked. If the tokens have not been used they will not appear in the list of currently valid tokens though they will still work if used unless the revoke all action is used. These tokens, if used, also count towards maximum token quota.

Reviewer 2: Is this info still relevant (the date states prior 2018... 4 years ago)

Author: AFAIU, although the tokens might be generated prior to 2018, they can still be in use. This text mentions that if such tokens exist, they contribute to the maximum token quota. If we can confirm that no such token exists anymore, we can safely remove this cautionary text. WDYT?

Reviewer 2: I believe we can ask in the #ask-platform slack channel if such tokens are still existing, the info is 4 years old, I expect almost all tokens to be expired/recreated by then. We can also leave it like this as of now and do the investigation on the side and eventually remove once we're sure the tokens are non existing anymore
